cmd/ncproxy/main.go

```diff
@@ -0,0 +1,117 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/Microsoft/hcsshim/cmd/ncproxy/nodenetsvc"
+	"github.com/Microsoft/hcsshim/internal/computeagent"
+	"github.com/Microsoft/hcsshim/internal/log"
+	"google.golang.org/grpc"
+)
+
+type nodeNetSvcConn struct {
+	client   nodenetsvc.NodeNetworkServiceClient
+	addr     string
+	grpcConn *grpc.ClientConn
+}
+
+var (
+	configPath = flag.String("config", "", "Path to JSON configuration file.")
+	// Global mapping of network namespace ID to shim compute agent ttrpc service.
+	containerIDToShim = make(map[string]computeagent.ComputeAgentService)
+	// Global object representing the connection to the node network service that
+	// ncproxy will be talking to.
+	nodeNetSvcClient *nodeNetSvcConn
+	// NcProxy configuration
+	conf = new(config)
+)
+
+func main() {
+	flag.Parse()
+	ctx := context.Background()
+	localConf, err := loadConfig(*configPath)
+	if err != nil {
+		log.G(ctx).WithError(err).Error("failed getting configuration file")
+		os.Exit(1)
+	}
+
+	conf = localConf
+	if conf.GRPCAddr == "" {
+		log.G(ctx).Error("missing GRPC endpoint in config")
+		os.Exit(1)
+	}
+
+	if conf.TTRPCAddr == "" {
+		log.G(ctx).Error("missing TTRPC endpoint in config")
+		os.Exit(1)
+	}
+
+	if conf.Timeout == 0 {
+		conf.Timeout = 5
+	}
+
+	// If there's a node network service in the config, assign this to our global client.
+	if conf.NodeNetSvcAddr != "" {
+		log.G(ctx).Debugf("connecting to NodeNetworkService at address %s", conf.NodeNetSvcAddr)
+
+		client, err := grpc.Dial(
+			conf.NodeNetSvcAddr,
+			grpc.WithInsecure(),
+			grpc.WithBlock(),
+			grpc.WithTimeout(time.Duration(conf.Timeout)*time.Second),
+		)
+		if err != nil {
+			log.G(ctx).Warnf("failed to connect to NodeNetworkService at address %s", conf.NodeNetSvcAddr)
+			os.Exit(1)
+		}
+
+		log.G(ctx).Debugf("successfully connected to NodeNetworkService at address %s", conf.NodeNetSvcAddr)
+
+		netSvcClient := nodenetsvc.NewNodeNetworkServiceClient(client)
+		nodeNetSvcClient = &nodeNetSvcConn{
+			addr:     conf.NodeNetSvcAddr,
+			client:   netSvcClient,
+			grpcConn: client,
+		}
+	}
+
+	log.G(ctx).Info("starting ncproxy")
```

Comment:
do we want this or any addition info to log here?

Reply:
Kind of just wanted this is a timestamp for when we tried to start running ncproxy, the individual services will both log their addrs and if they successfully started in `server.setup`

Reply:
Is there anything in addition you'd want to see logged here/remove this?

Reply:
Maybe including the config would be nice so that we can see in the logs the options that were set

Reply:
Good call!